src/main/java/tech/ypsilon/bbbot/discord/DiscordController.java

```diff
@@ -1,51 +1,77 @@
 package tech.ypsilon.bbbot.discord;
 
+import lombok.Getter;
 import net.dv8tion.jda.api.JDA;
 import net.dv8tion.jda.api.JDABuilder;
 import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.TextChannel;
 import net.dv8tion.jda.api.requests.GatewayIntent;
 import net.dv8tion.jda.api.utils.cache.CacheFlag;
-import tech.ypsilon.bbbot.settings.SettingsController;
+import tech.ypsilon.bbbot.ButterBrot;
+import tech.ypsilon.bbbot.util.GenericController;
+import tech.ypsilon.bbbot.util.Initializable;
 
 import javax.security.auth.login.LoginException;
 import java.util.Arrays;
 import java.util.Collection;
+import java.util.Objects;
 
-public class DiscordController {
+public class DiscordController extends GenericController implements Initializable {
 
     private static DiscordController instance;
 
-    private final JDA jda;
+    private @Getter JDA jda;
+    private @Getter Guild home;
+    private @Getter TextChannel logChannel;
 
     /**
      * Registering the JDA with the needed GatewayIntents.
      * @throws LoginException when the provided token is invalid
      */
-    public DiscordController() throws LoginException {
+    public DiscordController(ButterBrot parent) throws LoginException {
+        super(parent);
         instance = this;
-        Collection<GatewayIntent> gatewayIntents = Arrays.asList(GatewayIntent.GUILD_VOICE_STATES,
-                GatewayIntent.GUILD_MESSAGES, GatewayIntent.GUILD_MESSAGE_REACTIONS, GatewayIntent.DIRECT_MESSAGES,
+    }
+
+    @Override
+    public void init() {
+        Collection<GatewayIntent> gatewayIntents = Arrays.asList(
+                GatewayIntent.GUILD_VOICE_STATES,
+                GatewayIntent.GUILD_MESSAGES,
+                GatewayIntent.GUILD_MESSAGE_REACTIONS,
+                GatewayIntent.DIRECT_MESSAGES,
                 GatewayIntent.GUILD_PRESENCES
         );
 
-        jda = JDABuilder.createDefault((String) SettingsController.getValue("discord.token"), gatewayIntents)
-                .disableCache(CacheFlag.EMOTE, CacheFlag.ACTIVITY, CacheFlag.CLIENT_STATUS)
-                .build();
+        try {
+            jda = JDABuilder.createDefault(getParent().getConfig().getDiscord().getDiscordBotToken(), gatewayIntents)
+                    .disableCache(CacheFlag.EMOTE, CacheFlag.ACTIVITY, CacheFlag.CLIENT_STATUS)
+                    .build();
+
```

Comment:
You need to await jda to be ready. Otherwise the statements below can (and will very likely) evaluate to `null`.

Reply:
see c48b130